engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -792,36 +1026,50 @@ public BlockRegion addExtents(int extentX, int extentY, int extentZ, BlockRegion
      * @return dest
      */
     public BlockRegion addExtents(float extentX, float extentY, float extentZ, BlockRegion dest) {
-        dest.aabb.minX = Math.roundUsing(this.aabb.minX - extentX, RoundingMode.FLOOR);
-        dest.aabb.minY = Math.roundUsing(this.aabb.minY - extentY, RoundingMode.FLOOR);
-        dest.aabb.minZ = Math.roundUsing(this.aabb.minZ - extentZ, RoundingMode.FLOOR);
+        dest.minX = Math.roundUsing(this.minX - extentX, RoundingMode.FLOOR);
+        dest.minY = Math.roundUsing(this.minY - extentY, RoundingMode.FLOOR);
+        dest.minZ = Math.roundUsing(this.minZ - extentZ, RoundingMode.FLOOR);
+
+        dest.maxX = Math.roundUsing(this.maxX + extentX, RoundingMode.CEILING);
+        dest.maxY = Math.roundUsing(this.maxY + extentY, RoundingMode.CEILING);
+        dest.maxZ = Math.roundUsing(this.maxZ + extentZ, RoundingMode.CEILING);
 
-        dest.aabb.maxX = Math.roundUsing(this.aabb.maxX + extentX, RoundingMode.CEILING);
-        dest.aabb.maxY = Math.roundUsing(this.aabb.maxY + extentY, RoundingMode.CEILING);
-        dest.aabb.maxZ = Math.roundUsing(this.aabb.maxZ + extentZ, RoundingMode.CEILING);
         return dest;
     }
 
```

Comment:
I'm pretty sure this logic is incorrect @4Denthusiast

Reply:
Agreed (and I've added a test for that). But I'm also not sure whether there's a value in this method anymore, now that we treat integer coordinates as block centers. I think the following should hold (and could be the implementation):

```java
public BlockRegion addExtents(float extentX, float extentY, float extentZ, BlockRegion dest) {
	return addExtents(Math.roundUsing(extentX, RoundingMode.FLOOR), 
		Math.roundUsing(extentY, RoundingMode.FLOOR), 
		Math.roundUsing(extentZ, RoundingMode.FLOOR), dest);
}
```